### PR TITLE
Obfuscated prototype symbols for RC4

### PR DESCRIPTION
**Pull request by stevefan1999**
Since I don't know how to use the dependency injection system, I decided to reinvent the wheel in the `Utils` static class.

Fixes https://github.com/javascript-obfuscator/javascript-obfuscator/issues/159

### CODE REVIEW

File: src/utils/Utils.ts [hunk 1]
```diff
@@ -71,4 +71,24 @@ export class Utils {
             })
             .join(' + ');
     }
+    
+    /**
+     * @returns {string}
+     */
+    public static generateIden (): string {
```
Comment (sanex3339) [inline src/utils/Utils.ts]: You don't need this. You can use method `getRandomString` of `RandomGenerator` class.
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/utils/RandomGenerator.ts#L131

File: src/templates/string-array-nodes/string-array-calls-wrapper/SelfDefendingTemplate.ts [hunk 2]
```diff
@@ -1,46 +1,59 @@
 import { IEscapeSequenceEncoder } from '../../../interfaces/utils/IEscapeSequenceEncoder';
+import { Utils } from '../../../utils/Utils';
```
Comment (sanex3339) [inline src/templates/string-array-nodes/string-array-calls-wrapper/SelfDefendingTemplate.ts]: Check this line as example how to pass argument to template. In the example below `escapeSequenceEncoder` is passed to the template.
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/custom-nodes/self-defending-nodes/SelfDefendingUnicodeNode.ts#L73

And then in the template you can get it in the parameter:
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/templates/self-defending-nodes/self-defending-unicode-node/SelfDefendingTemplate.ts#L10

File: src/templates/string-array-nodes/string-array-calls-wrapper/SelfDefendingTemplate.ts [hunk 3]
```diff
@@ -1,46 +1,59 @@
...
 
 /**
  * @param {IEscapeSequenceEncoder} escapeSequenceEncoder
  * @returns {string}
  */
 export function SelfDefendingTemplate (escapeSequenceEncoder: IEscapeSequenceEncoder): string {
+    const symTbl: { [key: string]: string } = {
```
Comment (sanex3339) [inline src/templates/string-array-nodes/string-array-calls-wrapper/SelfDefendingTemplate.ts]: Please, use full names

File: src/templates/string-array-nodes/string-array-calls-wrapper/StringArrayRC4DecodeNodeTemplate.ts [hunk 4]
```diff
@@ -1,30 +1,39 @@
+import { Utils } from '../../../utils/Utils';
+
 /**
  * @returns {string}
  */
 export function StringArrayRc4DecodeNodeTemplate (): string {
+    const symTbl: { [key: string]: string } = {
```
Comment (sanex3339) [inline src/templates/string-array-nodes/string-array-calls-wrapper/StringArrayRC4DecodeNodeTemplate.ts]: Please use full name

File: src/templates/string-array-nodes/string-array-calls-wrapper/StringArrayRC4DecodeNodeTemplate.ts [hunk 5]
```diff
@@ -1,30 +1,39 @@
+import { Utils } from '../../../utils/Utils';
```
Comment (sanex3339) [inline src/templates/string-array-nodes/string-array-calls-wrapper/StringArrayRC4DecodeNodeTemplate.ts]: Check this line as example how to pass argument to template. In the example below `escapeSequenceEncoder` is passed to the template.
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/custom-nodes/self-defending-nodes/SelfDefendingUnicodeNode.ts#L73

And then in the template you can get it in the parameter:
https://github.com/javascript-obfuscator/javascript-obfuscator/blob/master/src/templates/self-defending-nodes/self-defending-unicode-node/SelfDefendingTemplate.ts#L10

**Comment by sanex3339** [comment]
I'll fix all comments by myself

**Comment by sanex3339** [comment]
Thank you for you PR, i'll change some parts of your code, but keep the idea of PR.

**Comment by stevefan1999** [comment]
@sanex3339 Oh dear I'm very sorry! I was busy learning UE4 at the moment...I forgot to push the changes. 

Anyway, I have seen you done that as well, appreciate it.

**Comment by sanex3339** [comment]
Yep, i fixed some issues in separate PR and will merge into master shortly.

**Pull request merged**